enhancements/installer/telemeter-metrics.md

```diff
@@ -0,0 +1,300 @@
+
+## Proposal
+
+### User Stories
+
+
+#### Cluster Admin/Installer User Stories
+As an admin about to install an OpenShift cluster I want to be able to disable metrics collection so that I can opt out.
+
+
+As an admin installing an OpenShift cluster I want my installation result to be unaffected by the success or failure of collecting metrics. 
+
+#### Telemetry Analyst
+Some examples:
+
+
+As a telemetry analyst I would like to create queries which help determine how long the installation process takes in total and broken down by different stages of installation so that I can better understand user experience and expose any abnormalities.
+
+
+As a telemetry analyst I would like to create queries which show aggregate counts of failure by stage of installation so that we can better understand where users are experiencing problems.
+
+
+As a telemetry analyst I would like to see how often users modify ignition configs or manifests so that I can better understand how users are interacting with the installer. [(see open question #4)](open-questions)
+
+
+As a telemetry analyst I would like to see how often the installer is being run on certain operating systems so I can better understand our user environments and needs.
+
+### Implementation Details/Notes/Constraints
+The installer does not fit a typical Prometheus use case in that:
+* metrics will be pushed rather than scraped
+* meaning will come from metric labels as much as values
+* the aggregation push gateway will be used to form the metrics into a useful time series
+
+Initially, three proposed metrics could provide value:
+
+#### cluster_installation_invocation
+The primary metric `cluster_installation_invocation` would be pushed when users invoke either the `create`, `destroy`, `gather`, or `wait-for` command. The values for the labels would be collected throughout the execution of the command and converted to a metric before being pushed.
+```
+# HELP cluster_installation_invocation represents a single command run by the OpenShift installer and related outcomes.
+# TYPE cluster_installation_invocation counter
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="25"} 1
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="ClusterComplete",return="0",version="4.2",duration="22.5"} 1
+cluster_installation_invocation{command="create", target="cluster",os="linux",platform="AWS",result="APIFailure",return="1",version="4.2",duration="30"} 1
+cluster_installation_invocation{command="destroy", target="cluster",os="linux",platform="AWS",result="Success",return="1",version="4.2",duration="10"} 1
+```
+* `duration` of the command represnts the overall time required to complete the command. The label values will be bounded in the installer client to preset buckets of a certain step-size, 2.5 in the example above.
```

Comment:
s/represnts/represents/

Reply:
Rather than encoding the duration bucket in the installer client, it might be more flexible/maintainable to perform the bucketing at the aggregation gateway level. Otherwise it will be impossible to adjust the buckets after the fact.

Reply:
```suggestion
* `duration` of the command represents the overall time required to complete the command. The label values will be bounded in the installer client to preset buckets of a certain step-size, 2.5 in the example above.
```

Reply:
And again, I think wit should be histogram as @brancz [mentioned](https://github.com/openshift/enhancements/pull/205/files#r380674219)